tests/010-test-use-devs-to-create-docker-root-volume.sh

```diff
@@ -0,0 +1,66 @@
+source $SRCDIR/libtest.sh
+
+# Test DOCKER_ROOT_VOLUME= directive. Returns 0 on success and 1 on failure.
+test_docker_root_volume() {
+  local devs=$TEST_DEVS
+  local test_status=1
+  local testname=`basename "$0"`
+  local vg_name="dss-test-foo"
+  local docker_root_lv_name="docker-root-lv"
+
+  # Error out if any pre-existing volume group vg named dss-test-foo
+  if vg_exists "$vg_name"; then
+    echo "ERROR: $testname: Volume group $vg_name already exists." >> $LOGS
+    return $test_status
+  fi
+
+  # Create config file
+  cat << EOF > /etc/sysconfig/docker-storage-setup
+DEVS="$devs"
+VG=$vg_name
+DOCKER_ROOT_VOLUME=yes
+DOCKER_ROOT_VOLUME_SIZE=40%FREE
+EOF
+
+ # Run docker-storage-setup
+ $DSSBIN >> $LOGS 2>&1
+
+ # Test failed.
+ if [ $? -ne 0 ]; then
+    echo "ERROR: $testname: $DSSBIN Failed." >> $LOGS
+    cleanup $vg_name "$devs"
+    return $test_status
+ fi
+
+  # Make sure $DOCKER_ROOT_VOLUME {docker-root-lv} got created
+  # successfully.
+  if ! lv_exists "$docker_root_lv_name"; then
+    echo "ERROR: $testname: Logical Volume $docker_root_lv_name does not exist." >> $LOGS
+    cleanup $vg_name "$devs"
+    return $test_status
+  fi
+
+   # Make sure $DOCKER_ROOT_VOLUME {docker-root-lv} is
+   # mounted on /var/lib/docker
+   local mnt
+   mnt=$(findmnt -n -o TARGET --first-only --source /dev/${vg_name}/${docker_root_lv_name})
+   if [ "$mnt" == "/var/lib/docker" ];then
+      test_status=0
+   fi
+
+  $DSSBIN --reset >> $LOGS 2>&1
+  # Test failed.
+  if [ $? -eq 0 ]; then
+     if [ ! -e /etc/sysconfig/docker-storage ]; then
+          test_status=0
```

Comment:
Ok, how about following. Write two tests.
- First tests to test volume creation funcitonality
- second test to test reset funcitonality.

In first test you can still use --reset to cleanup but don't use its return code to determine test status. This will be something like `cleanup $vg_name`. It can fail too but we don't determine success/failure of tests based on cleanup.

So first test will ignore the --reset results. It will PASS/FAIL based on whether volume creation was successful or not.

Second test will fail if reset did not work properly.